deployer/deployer-services/src/main/java/io/hyscale/deployer/services/builder/AppMetadataBuilder.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.deployer.services.builder;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.utils.ResourceLabelUtil;
+import io.hyscale.deployer.core.model.AppMetadata;
+import io.kubernetes.client.openapi.models.V1Pod;
+
+@Component
+public class AppMetadataBuilder {
+
+    /**
+     * Gets app and service name from labels generated by hyscale
+     * For pods not deployed with hyscale AppMetadata with only namespace will be returned
+     * @param podList
+     * @return List of {@link AppMetadata} containing details of deployed apps
+     */
+    public List<AppMetadata> listApplications(List<V1Pod> podList) {
+        if (podList == null) {
+            return null;
+        }
+        Map<String, AppMetadata> mapping = new HashMap<String, AppMetadata>();
+
+        podList.forEach(pod -> {
+            String namespace = pod.getMetadata().getNamespace();
+            String appName = ResourceLabelUtil.getAppName(pod.getMetadata().getLabels());
+            String serviceName = ResourceLabelUtil.getServiceName(pod.getMetadata().getLabels());
+
+            if (mapping.get(namespace) == null) {
```

Comment:
getorDefault

Reply:
This might not work as we would need to update the map as well

Reply:
Fetching the object from map only
try to use getOrDefault method instead of simple get or else do a contains check and then do a get